### PR TITLE
Chore: APP-866 Refactor Proposal Detail Page

### PR DESCRIPTION
**Pull request by Fabricevladimir**
## Description
- Extracts proposal mapping from subgraph into `useDaoProposal` hook

Note: Adds no functionality; setting the stage for upcoming PR

Task: [APP-866](https://aragonassociation.atlassian.net/browse/APP-866)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran all tests with success and extended them if necessary.
- [ ] I have updated the ``CHANGELOG.md`` file in the root folder.
- [x] I have tested my code on the test network.

### CODE REVIEW

**Review by RakeshUP — APPROVED** [review]

File: packages/web-app/src/pages/proposal.tsx [hunk 1]
```diff
@@ -7,91 +7,64 @@ import {
...
-import ResourceList from 'components/resourceList';
-import {StyledEditorContent} from 'containers/reviewProposal';
-import {VotingTerminal} from 'containers/votingTerminal';
-import {useMappedBreadcrumbs} from 'hooks/useMappedBreadcrumbs';
-import {useNetwork} from 'context/network';
-import {ERC20VOTING_PROPOSAL_DETAILS} from 'queries/proposals';
-import {NotFound} from 'utils/paths';
-import {useDaoParam} from 'hooks/useDaoParam';
+import useScreen from 'hooks/useScreen';
 import {Loading} from 'components/temporary';
```
Comment (WissenIstNacht) [inline packages/web-app/src/pages/proposal.tsx:23]: You're using VS Code, right? Could you do me a biggie and: 1) right click anywhere on this file. 2) Select Source Acion 3) Sort Import? 

I know it's not the optimal strategy. Eventually we should get eslint or something to do that. But turning it on all at once might be a bit too much. Well, I guess it depends whether you like your bandaid ripped up really quickly or really slowly haha

File: packages/web-app/src/hooks/useDaoProposal.tsx [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+};
+
+const publishedDone: ProgressStatusProps = {
+  label: 'Published',
+  mode: 'done',
+  date: '2021/11/16 4:30 PM UTC+2',
+  block: '132,123,231',
+};
+
+const Votes: StringIndexed = {
```
Comment (WissenIstNacht) [inline packages/web-app/src/hooks/useDaoProposal.tsx:37]: Isn't this losing information...? I mean the values here are `any`, whereas they could be `string` if you'd let TS just infer the types...?

**Review by WissenIstNacht — APPROVED** [review]
Nice! Thanks for doing this separately.

File: packages/web-app/src/hooks/useDaoProposal.tsx (hunk 2 above)
Comment (Fabricevladimir) [reply packages/web-app/src/hooks/useDaoProposal.tsx:37]: This is mostly for when accessing an object using a string as index. It's not *specifically* for this one.

**Pull request merged**